unittests/AST/ASTImporterTest.cpp

```diff
@@ -193,11 +193,23 @@ class ASTImporterTestBase : public ::testing::TestWithParam<ArgVector> {
     std::string FileName;
     std::unique_ptr<ASTUnit> Unit;
     TranslationUnitDecl *TUDecl = nullptr;
+    std::unique_ptr<ASTImporter> Importer;
     TU(StringRef Code, StringRef FileName, ArgVector Args)
         : Code(Code), FileName(FileName),
           Unit(tooling::buildASTFromCodeWithArgs(this->Code, Args,
                                                  this->FileName)),
-          TUDecl(Unit->getASTContext().getTranslationUnitDecl()) {}
+          TUDecl(Unit->getASTContext().getTranslationUnitDecl()) {
+      Unit->beginSourceFile();
+    }
+    
+    Decl *import(std::unique_ptr<ASTUnit> &ToAST, Decl *FromDecl) {
```

Comment:
I prefer taking a raw pointer than taking a reference to a unique pointer. On the call site you can use get, to get the underlying raw pointer.

Reply:
I would prefer to use reference (`ASTUnit &`) here, this makes it visible that the value should not be null. (But this practice is not mainly used in clang code.)

Reply:
Reference is also fine by me. This practice is being used in some parts of clang.